### PR TITLE
feat(core): return response from `loginWith`

### LINKED ISSUE #144: loginWith promise result is undefined
### Version

[v4.1.0](https://github.com/nuxt-community/auth-module/releases/tag/v4.1.0)

### Reproduction link

[https://nuxt-auth.herokuapp.com/](https://nuxt-auth.herokuapp.com/)

### Steps to reproduce

The following code logs undefined:

    
            return this.$auth.loginWith('local', {
                    data: {
                        username: this.username,
                        password: this.password
                    }
                }).then(response => {
                    console.log(response);
                }

### What is expected?

Getting the full response

### What is actually happening?

Response is undefined

> Moved from https://github.com/nuxt/consola/issues/11

<div align="right"><sub><em>This question is available on <a href="https://nuxtjs.cmty.io">Nuxt.js</a> community (<a href="https://nuxtjs.cmty.io/nuxt-community/auth-module/issues/c108">#c108</a>)</em></sub></div>

### LINKED ISSUE #249: loginWith/Login returning user object in promise
### What problem does this feature solve?

This would make it easy for devs to create the user in their own data store in e.g.
`.then({ })`
and errors in `.catch({ })`

so that means i should be able to do something like:
```JS
this.$auth.loginWith('facebook').then({
// Create the user in my own data store, e.g. MySQL etc.
}).catch({
// If the user creation fails in own data store, remove the data in vuex
await this.$auth.logout()
})
```

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/auth-module/issues/c206">#c206</a>)</em></sub></div>

### LINKED ISSUE #411: I can't get the result (response) by loginWith method
### What problem does this feature solve?

local schemes ignore the result except token, but sometimes we may need the hole response. For my situation, server will tell me some information about this login action, but I cant get it

### What does the proposed changes look like?

return the whole response

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/auth-module/issues/c375">#c375</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by JoaoPedroAS51**
Fix issue #144 

This closes #144, closes #411 and closes #249

### CODE REVIEW

**Comment by vincent-kean** [comment]
Waiting for this

File: docs/schemes/local.md [hunk 1]
```diff
@@ -83,3 +84,9 @@ This option can be used to disable all token handling. Useful for Cookie only fl
 - Default: `Bearer`
 
  Authorization header type to be used in axios requests.
+ 
+ ### `autoFetchUser`
```
Comment (pi0) [inline docs/schemes/local.md]: Would you please create a separate PR for this feature?

File: lib/core/auth.js [hunk 2]
```diff
@@ -128,10 +128,12 @@ export default class Auth {
       return Promise.resolve()
     }
 
-    return this.wrapLogin(this.strategy.login(...arguments)).catch(error => {
-      this.callOnError(error, { method: 'login' })
-      return Promise.reject(error)
-    })
+    return this.wrapLogin(this.strategy.login(...arguments))
+      .then(response => response)
```
Comment (pi0) [inline lib/core/auth.js]: What's the purpose of this line?

File: docs/schemes/local.md (hunk 1 above)
Comment (JoaoPedroAS51) [reply docs/schemes/local.md]: @pi0 Sure!

File: lib/core/auth.js (hunk 2 above)
Comment (JoaoPedroAS51) [reply lib/core/auth.js]: You're right! Just tested and it is not needed.

Commit https://github.com/nuxt-community/auth-module/pull/541/commits/23040007bae714405712c3aba7221a78c83a0a60

File: docs/schemes/local.md (hunk 1 above)
Comment (JoaoPedroAS51) [reply docs/schemes/local.md]: Revert commits: https://github.com/nuxt-community/auth-module/pull/541/commits/223a59b46d07d44d23673d4081c1ef1f4a0b7b76 and https://github.com/nuxt-community/auth-module/pull/541/commits/7d0bc40360fd2c2de2e1ecf793bbf7f7487d0155
autoFetchUser PR: https://github.com/nuxt-community/auth-module/pull/543

**Comment by pi0** [comment]
Awesome @JoaoPedroAS51 thanks!

**Comment by pi0** [comment]
Released with [v4.9.0](https://github.com/nuxt-community/auth-module/releases/tag/v4.9.0).

**Pull request merged**